inc/saivlan.h

```diff
@@ -487,6 +487,170 @@ typedef enum _sai_vlan_stat_t
 
 } sai_vlan_stat_t;
 
+/**
+ * @brief VLAN Stack stage
+ */
+typedef enum _sai_vlan_stack_stage_t
+{
+    SAI_VLAN_STACK_STAGE_INGRESS,
+
+    SAI_VLAN_STACK_STAGE_EGRESS,
+
+} sai_vlan_stack_stage_t;
+
+/**
+ * @brief VLAN Stack action
+ *
+ * If the action is SAI_VLAN_STACK_ACTION_SWAP, the original vlan id of the frame will be replaced
+ * with new vlan id and the new vlan id will be placed in the original tag position.
+ * If the action is SAI_VLAN_STACK_ACTION_PUSH, the original vlan id of the frame will be kept if
+ * exists and the new vlan id will be added as the outer vlan tag.
+ * If the action is SAI_VLAN_STACK_ACTION_POP, the original vlan id of the frame will be removed
+ * if exists.
+ */
+typedef enum _sai_vlan_stack_action_t
+{
+    SAI_VLAN_STACK_ACTION_SWAP,
+
+    SAI_VLAN_STACK_ACTION_PUSH,
+
+    SAI_VLAN_STACK_ACTION_POP,
+
+} sai_vlan_stack_action_t;
+
+/**
+ * @brief Attributes for VLAN Stack
+ */
+typedef enum _sai_vlan_stack_attr_t
+{
+    /**
+     * @brief Start of attributes
+     */
+    SAI_VLAN_STACK_ATTR_START,
+
+    /**
+     * @brief Stage type
+     * If action type is PUSH, this should set to INGRESS
+     * If action type is POP, this should set to EGRESS
+     *
+     * @type sai_vlan_stack_stage_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_VLAN_STACK_ATTR_STAGE = SAI_VLAN_STACK_ATTR_START,
+
+    /**
+     * @brief Action type
+     *
+     * @type sai_vlan_stack_action_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_VLAN_STACK_ATTR_ACTION,
+
+    /**
+     * @brief Original Vlan ID of the inner or outer
+     * The action will perform on the packets whose vlan tag match the corresponding vlan ID
+     *
+     * @type sai_vlan_stacking_vid_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID,
```

Comment:
To avoid meta changes can we use the foll:

SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID_OUTER
SAI_VLAN_STACK_ATTR_ORIGINAL_VLAN_ID_INNER

Reply:
Done. Please check.